index.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const methodOverride = require('method-override');
+const cookieParser = require('cookie-parser');
+const logger = require('morgan');
+const path = require('path');
+const clucksRouter = require('./routes/clucks/clucksRouter');
+const knex = require('./db/client')
+
+const app = express();
+
+function timeDifference(current, previous) {
+
+    var msPerMinute = 60 * 1000;
+    var msPerHour = msPerMinute * 60;
+    var msPerDay = msPerHour * 24;
+    var msPerMonth = msPerDay * 30;
+    var msPerYear = msPerDay * 365;
+
+    var elapsed = current - previous;
+
+    if (elapsed < msPerMinute) {
+         return 'Just Now';   
+    }
+
+    else if (elapsed < msPerHour) {
+         return Math.round(elapsed/msPerMinute) + ' minutes ago';   
+    }
+
+    else if (elapsed < msPerDay ) {
+         return Math.round(elapsed/msPerHour ) + ' hours ago';   
+    }
+
+    else if (elapsed < msPerMonth) {
+        return Math.round(elapsed/msPerDay) + ' days ago';   
+    }
+
+    else if (elapsed < msPerYear) {
+        return Math.round(elapsed/msPerMonth) + ' months ago';   
+    }
+
+    else {
+        return Math.round(elapsed/msPerYear ) + ' years ago';   
+    }
+}
+
+app.use(express.urlencoded({extended: true}));
+app.use(cookieParser());
+app.use((req, res, next) => {
+    const username = req.cookies.username || '';
+    res.locals.username = username;
+    next();
+})
+app.use(logger('dev'));
+app.use(methodOverride((req, res) => {
+    if (req.body && req.body._method) {
+        return req.body._method
+    }
+}));
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.set('view engine', 'ejs');
+app.set('views', path.join(__dirname, 'views'));
+
+app.get('/', async (req, res) => {
+    const hashtags =
+    await knex('hashtags')
+    .orderBy('count', 'desc')
+    .returning('*');
+
+    knex('clucks')
+    .orderBy('created_at', 'desc')
+    .then(clucks => {
+        res.render('clucks/index', { clucks, timeDifference, hashtags })
+    })
+})
+
+app.get('/sign_in', (req, res) => {
+    res.render('sign_in')
+})
+
+app.post('/sign_in', (req, res) => {
+    const { username } = req.body;
+    res.cookie('username', username);
+    res.redirect('/')
+})
+
+// add sign out button in navbar
+app.post('/sign_out', (req, res) => {
+    res.clearCookie('username');
+    res.redirect('/')
+})
+
+app.use('/clucks', clucksRouter);
+
+app.listen(3000, () => {
```

Comment:
declare PORT at the global scope and assign it to 3000 then use that variable here. t hen in the log you can interpolate `PORT`

Reply:
Too bad didn't get the comment before the quiz lol. By the way, could you share details on what is wrong for my trending tags section code? For me, the function and the view work exactly like what is required in the quiz.

Reply:
It didn't work for me. I can check again and get back to you by Saturday.

Reply:
Thank you for updating the grade! I‘m also curious about the marking standard for styling. So for super team picker, the home work requires we style our application as identical as possible to the sample. But for this quiz, it requires we follow the wireframes but we can make more design efforts. It has a marking section for design effort. I even asked the instructor on quiz day, whether we should make the app prettier based on the wireframe, or we should make it exactly the same as the sample. He said we should make it prettier. I think I did make the app prettier using my own style sheet with bootstrap haha. But I got reduction for styling. I don’t know it’s because the miscommunication about the styling marking standard, or my design is ugly😂. Just a tiny thing. I know functionality is the key. But I do want to clarify with the marking standard for styling so I can get better for the following assignments. Thank you and have a great weekend!😊

Reply:
I understand and I think things were not communicated properly. I'm marking based on the rubrics and it focuses on how "identical" yours is to the mocks. If I have to list out the details, I usually focus on:
* does the mocks have nested elements? e.g. a big rectangle wrapper containing sub rectangles, titles and buttons, etc
* how close is the margin/padding/spacing
* i don't care much about the colour, image. you have a freedom to choose any font but nothing too crazy.

Since you asked the instructor and your app was indeed pretty and I can see the effort you have put in, I can give you back those lost points on styling. 
Not sure if I will be marking HW5 & 6 but if I do, I will let you know. If you don't hear back from me by mid next week then just assume I won't be the one marking it haha. If I do happen to mark those, the styling mark will be based off how identical the app is to the mocks :)

Reply:
I see I see. Thank you for sharing these details! I appreciate it! I’ll try to make hw5 identical to the mocks haha